src/components/DayPickerRangeController.jsx

```diff
@@ -160,6 +160,8 @@ export default class DayPickerRangeController extends React.Component {
     if (focusedInput !== END_DATE) return false;
 
     if (startDate) {
+      if (minimumNights === 1) return isSameDay(day, startDate);
```

Comment:
So I think the helper methods (`isSameDay`, etc.) all compare both dates at midnight. If we were to change the `dayDiff` line to:
```
      const dayDiff = day.clone().startOf('day').diff(startDate.clone().startOf('day'), 'days');
```
that should work.